### PR TITLE
Here valid range

### LINKED ISSUE #554: Create a table in `here` that documents which ta data corresponds to which map version
And then update the documentation.

### PR DESCRIPTION
**Pull request by chmnata**
## What this pull request accomplishes:

- create a valid range table in schema here
- update readme to point to that table

## Issue(s) this solves:

- closes #554 

## What, in particular, needs to reviewed:

- the readme 
- and the sql

### CODE REVIEW

File: here/README.md [hunk 1]
```diff
@@ -117,7 +117,7 @@ JOIN here_gis.streets_att_16_1 gis ON gis.link_id = LEFT(ta.link_dir, -1)::numer
 
 HERE updates their maps on an annual basis (usually) to keep up with Toronto's evolving street network. It's important to make sure that the traffic analytics data you're using matches the street network. 
 
-Since August 2022, the `here.ta` table corresponds to the `here.routing_streets_22_2` layer. You can check to see if the `here.ta` layer corresponds to any given map layer by joining the attributes using the `link_dir` field and examining the results.
+Please refer to table `here.street_valid_range` for which street version correspond to what range of data in `here.ta`. This table updates annually along with the HERE map refresh.
```
[the remark this answers is not in the document]
Comment (chmnata) [reply here/README.md:120]: I'm gonna move it under `HERE Data: dates and time periods`

**Review by radumas — DISMISSED** [review]
See previous comment regarding documentation

**Comment by chmnata** [comment]
@radumas ready for re-review, moved the section more to the front

File: here/traffic/sql/create_here_valid_range.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+-- which date range in here.ta table 
+
+CREATE TABLE here.street_valid_range AS 
+
+WITH street_v(street_version, valid_range) AS (
+	VALUES 
+	('18_3', daterange('2014-01-01', '2017-01-01')),
+	('19_4_tc', daterange('2017-01-01', '2017-09-01')),
+	('21_1', daterange('2017-09-01', '2022-08-15')),
+	('22_2', daterange('2022-08-15', 'infinity'))
```
Comment (Nate-Wessel) [inline here/traffic/sql/create_here_valid_range.sql:11]: Better to keep the upper range on this one undefined. https://www.postgresql.org/docs/current/rangetypes.html#RANGETYPES-INFINITE

Should be the same effect, but clearly indicates that this is a (currently) unknown value.

File: here/README.md [hunk 3]
```diff
@@ -60,6 +60,14 @@ We have HERE data going back to 2012, but there sure weren't as many people driv
...
 
+## Matching maps and data
+
+HERE updates their maps on an annual basis (usually) to keep up with Toronto's evolving street network. It's important to make sure that the traffic analytics data you're using matches the street network. This only applies when you're using the raw table `here.ta`.
+
+Refer to table `here.street_valid_range` for which street version correspond to what range of data in `here.ta`. This table updates annually along with the HERE map refresh.
+
+For example, if you are selecting speed data from 2017-09-01 to 2022-08-15, the correspinding street version is `21_1`. Relevant tables for your use case will have a `21_1` suffix, e.g. `here.routing_streets_21_1`.
```
Comment (Nate-Wessel) [inline here/README.md:69]: typo: "correspinding"

File: here/README.md [hunk 4]
```diff
@@ -60,6 +60,14 @@ We have HERE data going back to 2012, but there sure weren't as many people driv
 
 Though data are reported at 5-minute intervals, on any given link (aka road segment) there may not be a lot of observations. Longer time periods and roads with higher traffic volumes have more probe data, and therefore more accurate results, than short time periods on roads with light traffic. More probe data increases accuracy. There are some road segments in the City (cul-de-sacs and other lightly travelled roads) that have very few observations. Because of this, we calculate travel times for time periods that are at least three weeks or longer so that data from multiple days can be aggregated into a more accurate result.
 
+## Matching maps and data
+
+HERE updates their maps on an annual basis (usually) to keep up with Toronto's evolving street network. It's important to make sure that the traffic analytics data you're using matches the street network. This only applies when you're using the raw table `here.ta`.
+
+Refer to table `here.street_valid_range` for which street version correspond to what range of data in `here.ta`. This table updates annually along with the HERE map refresh.
```
Comment (Nate-Wessel) [inline here/README.md:67]: Where are the street data exactly? I'm just assuming this refers to `here_gis.streets_*` and `here_gis.streets_att_*`

**Review by Nate-Wessel — CHANGES_REQUESTED** [review]
Aren't there more than four street versions? I see:
* `streets_16_1`
* `streets_16_4`
* `streets_17_4`
* `streets_18_3`
* `streets_19_4`
* `streets_19_4_tc`
* `streets_21_1`
* `streets_22_2`

File: here/README.md (hunk 4 above)
Comment (chmnata) [reply here/README.md:67]: year it is just `here_gis.streets_*`

Comment (Nate-Wessel) [reply here/README.md:67]: Can you just make that explicit in the documentation?

**Comment by Nate-Wessel** [comment]
> Aren't there more than four street versions? I see:
> 
>     * `streets_16_1`
> 
>     * `streets_16_4`
> 
>     * `streets_17_4`
> 
>     * `streets_18_3`
> 
>     * `streets_19_4`
> 
>     * `streets_19_4_tc`
> 
>     * `streets_21_1`
> 
>     * `streets_22_2`

@chmnata can you address this? I'm not sure if these are tables that have no corresponding data in `here.ta` (if so, why do we have them?) or if their valid date ranges are just undefined here.

**Comment by chmnata** [comment]
> > Aren't there more than four street versions? I see:
> > ```
> > * `streets_16_1`
> > 
> > * `streets_16_4`
> > 
> > * `streets_17_4`
> > 
> > * `streets_18_3`
> > 
> > * `streets_19_4`
> > 
> > * `streets_19_4_tc`
> > 
> > * `streets_21_1`
> > 
> > * `streets_22_2`
> > ```
> 
> @chmnata can you address this? I'm not sure if these are tables that have no corresponding data in `here.ta` (if so, why do we have them?) or if their valid date ranges are just undefined here.

Added in the 17_4, I somehow thought we don't have them anymore 🤔 
I think whatever before 17_4 is not longer relevant cause back in the day when the map version changes, we re-pull previous data to use the updated map version (but it has its limits because you can only pull up to (i think 2 years?) of previous data)

**Comment by Nate-Wessel** [comment]
> Added in the 17_4, I somehow thought we don't have them anymore 🤔 I think whatever before 17_4 is not longer relevant cause back in the day when the map version changes, we re-pull previous data to use the updated map version (but it has its limits because you can only pull up to (i think 2 years?) of previous data)

So should that older data be removed? Or maybe somehow flagged as of-historical-interest only? Perhaps enter them here with a `null` range and some additional field with a comment?

**Comment by chmnata** [comment]
yea I think we can remove 16_4 and the ones before that

**Comment by chmnata** [comment]
Dropped all these
```
DROP TABLE here_gis.adminbndy2_16_1;
DROP TABLE here_gis.adminbndy2_16_4;
DROP TABLE here_gis.adminbndy3_16_1;
DROP TABLE here_gis.adminbndy3_16_4;
DROP TABLE here_gis.adminbndy4_16_1;
DROP TABLE here_gis.adminbndy4_16_4;
DROP TABLE here_gis.adminbndy5_16_1;
DROP TABLE here_gis.adminbndy5_16_4;

DROP TABLE here_gis.adminline1_16_1;
DROP TABLE here_gis.adminline1_16_4;
DROP TABLE here_gis.adminline2_16_1;
DROP TABLE here_gis.adminline2_16_4;

DROP TABLE here_gis.altstreets_16_1;
DROP TABLE here_gis.autosvc_16_1;
DROP TABLE here_gis.autosvc_16_4;

DROP TABLE here_gis.bordcross_16_1;
DROP TABLE here_gis.bordcross_16_4;

DROP TABLE here_gis.business_16_1;
DROP TABLE here_gis.business_16_4;

DROP TABLE here_gis.cartocountry_16_1;
DROP TABLE here_gis.cartocountry_16_4;

DROP TABLE here_gis.cartostate_16_1;
DROP TABLE here_gis.cartostate_16_4;

DROP TABLE here_gis.commsvc_16_1;
DROP TABLE here_gis.commsvc_16_4;

DROP TABLE here_gis.eduinsts_16_1;
DROP TABLE here_gis.eduinsts_16_4;

DROP TABLE here_gis.entertn_16_1;
DROP TABLE here_gis.entertn_16_4;

DROP TABLE here_gis.featpoint_16_1;
DROP TABLE here_gis.featpoint_16_4;

DROP TABLE here_gis.fininsts_16_1;
DROP TABLE here_gis.fininsts_16_4;

DROP TABLE here_gis.hospital_16_1;
DROP TABLE here_gis.hospital_16_4;

DROP TABLE here_gis.islands_16_1;
DROP TABLE here_gis.islands_16_4;

DROP TABLE here_gis.landmark_16_1;
DROP TABLE here_gis.landmark_16_4;

DROP TABLE here_gis.landuseb_16_1;
DROP TABLE here_gis.landuseb_16_4;

DROP TABLE here_gis.majhwys_16_1;
DROP TABLE here_gis.majhwys_16_4;

DROP TABLE here_gis.majhwyshield_16_1;
DROP TABLE here_gis.majhwyshield_16_4;

DROP TABLE here_gis.misccategories_16_1;
DROP TABLE here_gis.misccategories_16_4;

DROP TABLE here_gis.namedplc_16_1;
DROP TABLE here_gis.namedplc_16_4;

DROP TABLE here_gis.parking_16_1;
DROP TABLE here_gis.parking_16_4;

DROP TABLE here_gis.parkrec_16_1;
DROP TABLE here_gis.parkrec_16_4;

DROP TABLE here_gis.railrds_16_1;
DROP TABLE here_gis.railrds_16_4;

DROP TABLE here_gis.restrnts_16_1;
DROP TABLE here_gis.restrnts_16_4;

DROP TABLE here_gis.sechwys_16_1;
DROP TABLE here_gis.sechwys_16_4;

DROP TABLE here_gis.sechwyshield_16_1;
DROP TABLE here_gis.sechwyshield_16_4;

DROP TABLE here_gis.shopping_16_1;
DROP TABLE here_gis.shopping_16_4;

DROP TABLE here_gis.streets_16_1;
DROP TABLE here_gis.streets_16_4;

DROP TABLE here_gis.streets_att_16_1;
DROP TABLE here_gis.streets_att_16_4;

DROP TABLE here_gis.transhubs_16_1;
DROP TABLE here_gis.transhubs_16_4;

DROP TABLE here_gis.travdest_16_1;
DROP TABLE here_gis.travdest_16_4;

DROP TABLE here_gis.waterpoly_16_1;
DROP TABLE here_gis.waterpoly_16_4;

DROP TABLE here_gis.waterseg_16_1;
DROP TABLE here_gis.waterseg_16_4;

DROP TABLE here_gis.zlevels_16_1;
DROP TABLE here_gis.zlevels_16_4;
```

**Review by Nate-Wessel — APPROVED** [review]
Looks good to me now that the older tables are gone. Thanks!

**Pull request merged**